### DISCUSSION TITLE
Unable to run website on local network

### DISCUSSION THREAD

**Discussion by haleyngonadi** [laravel/valet#1143] [Q&A] [answered]
## Clear description of your problem
I have followed the instructions on the official website on how to run a website on a local network but I'm unable to access my .test website on my local network. 

## Expected behavior
Accessing my .test website via my IP address like so: 10.0.0.23/site.test and being able to see my work site.test website.

## Current behavior
The site (via IP address and via .test URL) first returns a privacy error and then when I skip it, I get a 404 Not Found Error as soon as I removed the prefix from listen 80 & 443 and restart valet. valet secure has been run in the directory.

## Steps to Reproduce

1. Remove the prefix on the listen directive for ports 80 and 443 in the Nginx config of the secured site. 
2. Restart Valet
3. Try accessing the website via ipaddress/site.test

**Comment by devinfd** [comment]
Same issue.

**Reply by fyodorovich** [reply]
I'm having a similar problem. Running on Big Sur with php 7.4, and everything up to date. Everything appears to be in place. I have linked a website that runs perfectly happily when I'm using Apache Netbeans to fire up a server with `php -S localhost:8080.` but it does not respond under valet. The ping test works on foobar.test but when I try to ping the linked site by name I get request timeouts.

**Comment by blakekrone** [comment]
Same here as well. I can access the sites just fine from the local machine. Switching the listen to be 0.0.0.0:80 I can hit the server from another machine on the network. If I add the site name after the IP it won't load. I can set a default site and it works just fine. FWIW I am using valet park.

**Comment by alexking** [comment]
The official documentation seems to be incorrect here, following the instructions on the latest version of valet gives a 404 on `ip/project.tld`. This should probably be converted back to an issue.

**Reply by haleyngonadi** [reply]
Same thing happening on my end!

**Comment by devinfd** [comment]
Maybe I don’t understand the difference between a discussion and an issue but to me this seems like it definitely is an issue sense the documented behavior and functionality doesn’t work.

**Comment by blakekrone** [comment]
Agreed, just seems like deflecting and making it appear as though there are fewer issues than there are. While I agree a discussion is warranted it has been recognized by multiple people which to me sounds like a bug.

**Comment by indieSoftware** [comment]
It seems it's not a bug, but a feature. With https://github.com/laravel/valet/pull/814 it's not intended to listen from outside of localhost, therefore, it seems it's not possible to share your local site with your local network. As a workaround you might use Ngrok https://laravel.com/docs/8.x/valet#sharing-sites-via-ngrok to tunnel your site to the outside-world which then can also be accessed by devices in your local network.

**Reply by haleyngonadi** [reply]
Interesting. How come that isn't mentioned in the official docs :/

**Reply by fyodorovich** [reply]
Perhaps I should not have placed my "me too" here. I'm working on localhost, not on the local network.

**Comment by brrrm** [comment]
As far as I know the only way to solve this is to set a "default" option in ~/.config/valet/config.json like so:

```
{
    "paths": [
        "/Users/USERNAME/.config/valet/Sites"
    ],
    "tld": "test",
    "loopback": "127.0.0.1",
    "default": "local/path/to/site"
}
```

**Comment by fgschafer** [comment]
But then?? no more access from local network? I installed valet SPECIFICALLY for that feature!!

**Comment by mattstauffer** [comment, marked as answer]
The official documentation shows how to use this feature:

https://laravel.com/docs/9.x/valet#sharing-sites-on-your-local-network

**Reply by wingsline** [reply]
Editing the config for the sites works until you run `valet install` (to upgrade of fix issues) and then all your modified nginx config files will be reverted. Not really a proper solution for sharing sites on the local network (or with any virtual machine). 

Also modifying the nginx files, valet gave me the wrong site's certificates after restart.

Only proper solution that worked is by modifying the stub files for valet and re-run valet install.
